Enhancements and fixes for the Authorization of Web Endpoints section (#30549)
quarkusio/quarkus: docs/src/main/asciidoc/security-authorization-of-web-endpoints-reference.adoc

https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
////

> [id="security-authorization-of-web-endpoints-reference"]

Reviewer: @MichalMaler - Was just about to link to this great page from the security overview PR and realized I missed this suggestion:

Would it make sense to reduce length of filename to:

 `security-authorize-web-endpoints-reference.adoc` 

and the respective ID:

```suggestion
[id="security-authorize-web-endpoints-reference.adoc"]
```

Author: @michelle-purcell Yeah, why not :) Sure. Do you want me to do it in this PR or we can merge and you will do it in yours? What is better in your books?

Author: Renamed the file, changed the ID and fixed the xrefs :)

Author: Renaming patch here:
https://github.com/quarkusio/quarkusio.github.io/pull/1616

Reviewer: @MichalMaler - It's your baby! ... Thanks. Looks great :+1:

TIP: Given the above permission set, `GET /api/foo` would match both permission sets' paths, requiring both the `user` and `admin` roles.

=== Configuration properties to deny access

The following configuration settings alter the role-based access control (RBAC) denying behavior:

`quarkus.security.jaxrs.deny-unannotated-endpoints=true|false`::

> If set to true, the access will be denied for all JAX-RS endpoints by default, so if a JAX-RS endpoint does not have any security annotations, it will default to the `@DenyAll` behavior.

Reviewer: Suggestion to reduce sentence length:

```suggestion
If set to true, access is denied for all JAX-RS endpoints by default. If a JAX-RS endpoint does not have any security annotations, it defaults to the `@DenyAll` behavior.
```
